### PR TITLE
Cleanup common ui elements, add test cases

### PR DESCRIPTION
**Pull request by guerler**

### CODE REVIEW

File: client/galaxy/scripts/mvc/ui/ui-misc.js [hunk 1]
```diff
@@ -10,257 +10,141 @@ define(['utils/utils',
...
-                title   : '',
-                cls     : '',
-                tagname : 'label'
-            });
-            this.setElement(this._template(this.options));
+        tagName: 'label',
+        initialize: function( options ) {
+            this.model = options && options.model || new Backbone.Model( options );
+            this.tagName = options.tagName || this.tagName;
+            this.setElement( $( '<' + this.tagName + '/>' ) );
```
Comment (carlfeberhard) [inline client/galaxy/scripts/mvc/ui/ui-misc.js:19]: You don't need the two above. 

```
v1 = Backbone.View({ tagName: 'label' }); 
v2 = Backbone.View({ tagName: 'li' }); 
```

will produce

```
v1.tagName === 'label'; // and v1.el will be <label></label>;
v2.tagName === 'li'; // and <li></li>
```

**Comment by carlfeberhard** [comment]
+1 for adding render to the lifecycle; overall looks fine.

I'm still not a fan of using Model like a data-binding/observable-object for UI states - but I think the ship has sailed on that.

**Comment by martenson** [comment]
:+1:

**Comment by guerler** [comment]
Awesome. Thanks, Carl. I'll remove those two lines with the next iteration.

**Pull request merged**